contrib/test.sh

```diff
@@ -2,14 +2,36 @@
 
 set -ex
 
+if [ -n "$1" ];
+then
+	deps="$1"
+else
+	deps="minimal recent"
+fi
+
 CRATES="bitcoin hashes internals"
 
-for crate in ${CRATES}
+for dep in $deps
 do
-    (
-        cd "$crate"
-        ./contrib/test.sh
-    )
+	cp Cargo-$dep.lock Cargo.lock
+	for crate in ${CRATES}
+	do
+	    (
+		cd "$crate"
+		./contrib/test.sh
+	    )
+	done
+	if [ $dep = recent ];
+	then
+		# We always test committed dependencies but we want to warn if they could've been updated
+		cargo update
+		if diff Cargo-recent.lock Cargo.lock;
+		then
+			echo Dependencies are up to date
+		else
+			echo "::warning file=Cargo-recent.lock::Dependencies could be updated"
```

Comment:
Is this like a format that GitHub CI will pick up and somehow make visible?

Reply:
This is some github syntax that gives color in their logs, @Kixunil found it but I didn't look up the docs when he told me about it.

Reply:
IIRC it's visible when you open the lock file from the PR. It was long time since I looked into it.